### DISCUSSION TITLE
Ordering heatmap columns

### DISCUSSION THREAD

**Discussion by KidneyNode** [david-barnett/microViz#116] [Q&A] [answered]
Is there a way to order heatmap columns by the sample name, rather than by a distance method?

This is my current code, from a phyloseq object called hm_T3_11

htmp_T3_11 <- hm_T3_11 %>%
  tax_transform("clr", zero_replace = "halfmin", rank = "Genus") %>%
  comp_heatmap(
    taxa = 1:30, samples = NA, colors = heat_palette(sym = TRUE), column_title = "11 - Baseline", name = "CLR",
    sample_anno = sampleAnnotation(
      Responder = anno_sample_cat("wuhan_response", legend_title = "Wuhan Respone", col = list("white","black"))),
    sample_names_show = FALSE,
    show_column_dend = FALSE
  )

**Comment by david-barnett** [comment, marked as answer]
Yes, you can set the sample_seriation argument to the "Identity" method, which keeps the samples in the original order.

You can sort the samples beforehand using ps_arrange and whichever variable(s) in the sample data you like, e.g.

```r
data("dietswap", package = "microbiome")
dietswap %>%
    ps_filter(timepoint == 2) %>% 
    tax_agg("Genus") %>% 
    tax_sort(by = sum) %>%
    ps_arrange(nationality, sample) %>% 
    comp_heatmap(taxa = 1:20, sample_seriation = "Identity", sample_names_show = TRUE)
```

**Reply by KidneyNode** [reply]
Brilliant! Thank you
